packages/components/src/theme-type-badge/premium-badge/index.tsx

```diff
@@ -46,10 +46,19 @@ const PremiumBadge = ( {
 		  );
 
 	const divRef = useRef( null );
+	const labelRef = useRef( null );
 	const [ isPopoverVisible, setIsPopoverVisible ] = useState( false );
 	const [ isHovered, setIsHovered ] = useState( false );
 	const [ isPressed, setIsPressed ] = useState( false );
 
+	// Display the label as a tooltip if the tooltip is being hidden and the label is too long.
+	const displayLabelAsTooltip =
```

Comment:
this needs to run when the user has hovered over the badge as only then will `labelRef.current.offsetWidth` et al be updated to the actual correct numbers. 

With the current structure, that means hovering over the badge needs to trigger a re-render, which a) I don't know how to do and b) seems like it might be a bad idea.